src/engine/engine.rs

```diff
@@ -36,17 +36,18 @@ pub trait KeyActions {
     /// by a newline.
     ///
     /// Successful return values:
-    /// * `Ok(CreateAction::Identity)`: The key was already in the keyring with the
-    /// appropriate key description and key data.
-    /// * `Ok(CreateAction::Created(false)`: The key was newly added to the keyring.
-    /// * `Ok(CreateAction::Created(true)`: The key description was already present
+    /// * `Ok(MappingCreateAction::Identity)`: The key was already in the keyring
+    /// with the appropriate key description and key data.
+    /// * `Ok(MappingCreateAction::Created(()))`: The key was newly added to the
+    /// keyring.
+    /// * `Ok(MappingCreateAction::Changed)`: The key description was already present
```

Comment:
You might want to make a note here the parameterizing on the ```()``` type is part of a security protocol.

Reply:
I'm not sure quite what you mean. Are you suggesting a security protocol section in the documentation?

Reply:
Disregard the previous comment, I misread your suggestion.

Reply:
So, I'm worried about somebody coming along and saying to themselves that the method would be so much more useful if it returned the mapping, not just ```()``` and making it so. So, a comment that says that this is a deliberate choice would be helpful here. I haven't dug into the rest of the code yet deeply, but I think part of the point is that it should be impossible. So, I think it should be worth a note right here.